### PR TITLE
Add proxy handling to HttpClientHelper.httpClientObjWithPerRequestTim…

### LINKED ISSUE #871: ServiceClient webproxy problem
See [link](https://github.com/Azure-Samples/azure-iot-samples-csharp/issues/41)

### PR DESCRIPTION
**Pull request by jenoola**
…eout

## Checklist
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

## Reference/Link to the issue solved with this PR (if any)

Fixes #871 
Fixes Azure-Samples/azure-iot-samples-csharp#41

### CODE REVIEW

File: common/src/service/HttpClientHelper.cs [hunk 1]
```diff
@@ -66,7 +66,17 @@ sealed class HttpClientHelper : IHttpClientHelper
...
+            {
+                HttpClientHandler httpClientHandler = new HttpClientHandler();
+                httpClientHandler.UseProxy = (customHttpProxy != null);
+                httpClientHandler.Proxy = customHttpProxy;
+                this.httpClientObjWithPerRequestTimeout = new HttpClient(httpClientHandler);
+            }
+            else
+            {
+                this.httpClientObjWithPerRequestTimeout = new HttpClient();
+            }
```
Comment (prmathur-microsoft) [inline common/src/service/HttpClientHelper.cs]: E2E for this flow

Comment (prmathur-microsoft) [reply common/src/service/HttpClientHelper.cs]: Add proxy test on the connectivity

File: common/src/service/HttpClientHelper.cs [hunk 2]
```diff
@@ -66,7 +66,17 @@ sealed class HttpClientHelper : IHttpClientHelper
             this.httpClientObj.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(CommonConstants.MediaTypeForDeviceManagementApis));
             this.httpClientObj.DefaultRequestHeaders.ExpectContinue = false;
 
-            this.httpClientObjWithPerRequestTimeout = new HttpClient();
+            if (customHttpProxy != DefaultWebProxySettings.Instance)
```
Comment (CIPop) [inline common/src/service/HttpClientHelper.cs]: @abhipsaMisra @jenoola please work together to identify what is missing from the existing serviceclient code: we already have proxy support for ServiceClient C2D (test should exist). as well as RegistryManager, JobClient, etc which work on HTTP. 

Needing to add this code here means we either missed something or the previous code was not added at the right depth in our stack.

File: e2e/test/DeviceStreamingE2ETests.cs [hunk 3]
```diff
@@ -45,6 +45,19 @@ public async Task DeviceStreaming_RequestAccepted_Sas_Amqp()
             await TestDeviceStreamingAsync(TestDeviceType.Sasl, transportSettings, true).ConfigureAwait(false);
         }
 
+        [TestMethod]
+        public async Task DeviceStreaming_RequestAccepted_Sas_Amqp_WithProxy()
+        {
+            Client.AmqpTransportSettings amqpTransportSettings = new Client.AmqpTransportSettings(Client.TransportType.Amqp_Tcp_Only)
```
Comment (CIPop) [inline e2e/test/DeviceStreamingE2ETests.cs]: The test doesn't match the code change above: this is setting AMQP proxy and above is HTTP.

**Comment by CIPop** [comment]
@jenoola #1029  contains the same change. Please separate the two items:
1. First check-in to master.
2. Cherry-pick into preview to ensure we track changes and not have 2 PRs for the exact same change.
3. Keep fixes separate - the other PR has 2 unrelated changes AFAIK.

**Comment by jenoola** [comment]
Upon further investigation it was determined httpClientObjWithPerRequestTimeout is used in preview but not in master. Hence, #1029 should be closed, dead code removed from master, and proxy support for httpClientObjWithPerRequestTimeout  will be added only to preview branch

**Pull request closed without merging**